### PR TITLE
[FIX] Rooms list sorting by activity multiple re-renders and case sensitive sorting alphabetically

### LINKED ISSUE #5341: Direct message conversations not sorted alphabetically
Right now users can create usernames with capital letters, which makes sorting results not so friendly, b/c instead of `['apple', 'Banana', 'Pineapple']`, we have `['Banana', 'Pineapple', 'apple']`. It's extremely confusing when you have opened like 30 chats all the time and everyone follows different naming convention (at all the best option will be to replace nicknames with real names, but AFAIK there is an open PR with this feature)

So we have few possible solutions:

Disclaimer: I'm not a back-end guy, I have no idea what's are good patterns here.

1. Force users to use usernames without capital letters - for me this is best option, b/c users have to keep usernames unique, even now we can't create `@user`,`@USER`, `@User` and `@UsEr`, so why not make this simpler? Also this will make everything more consistent, b/c channel names are already limited to lowercase and it works as expected.

2. Use `caseInsensitive` option for `ensureIndex` method, but it's available in MongoDB >= 3.3.11 ->  https://jira.mongodb.org/browse/SERVER-90
AFAIK Rocket.Chat didn't fit this requirements.

3. Store additional field in DB with name in lowercase and use it for sorting purposes only. I tried to make this, but without luck 😢

### LINKED ISSUE #9253: DMs are sorted asciibetically
### Description:
DMs in the room list are sorted asciibetically instead of alphabetically. This makes it harder to find users in a long DM list. Asciibetically means uppercase letters appear before lowercase letters, for example:

```
AUser1
BUser3
CUser5
auser2
buser4
cuser6
```

Should instead be sorted:

```
AUser1
auser2
BUser3
buser4
CUser5
cuser6
```

Asciibetical sorting can result in unnatural sorting in non-English languages, especially CJK ones, too. 

There are packages that provide natural sorting ([1](https://www.npmjs.com/package/node-natural-sort), [2](https://www.npmjs.com/package/natural-sort)). A naïve way to achieve similar results (generally for latin only) is to convert to lowercase while sorting.

### Server Setup Information:

* Version of Rocket.Chat Server: 0.60.0-develop (5de3446)
* Operating System: Ubuntu 16.04
* Deployment Method(snap/docker/tar/etc): Built from scratch, run under pm2
* Number of Running Instances: 2
* Node Version: 8.2.1

### Steps to Reproduce:

1. Create DMs with two users, where one's name starts with a lowercase letter, and the other starts with an uppercase letter

### Expected behavior:

1. List should be sorted alphabetically (AaBbCcDd...) instead of asciibetically (ABCDabcd...)

### Actual behavior:

1. List is sorted asciibetically (ABCDabcd...) instead of alphabetically (AaBbCcDd...)

### LINKED ISSUE #9295: [BUG] Sorting of usernames is case-sensitive
Usernames can be chosen as mixed case strings, which results in an order of usernames like following (sidebar):
A
B
C
a
b
c

They should be up or downcasted before sorting.

### LINKED ISSUE #10004: [Improvment] Sorting in private message list
### Description:

If you sort your private messages alphabetically, the sorting is not correct. Capital letters are ranked higher, as the following example shows.

### Server Setup Information:

* Version of Rocket.Chat Server: 0.62.1

### Steps to Reproduce:

1. Create two users: `aabc` and `Abc`
2. Start private conversations with them
3. Have a look in the left sidebar

### Expected behavior:

Ranking:
- aabc
- Abc

### Actual behavior:

Ranking:
- Abc
- aabc

Real world example in open.rocket.chat:

rocket_chat_

### LINKED ISSUE #10415: Do not close search sidebar on idle timeout
### Description:

There seem to be two related issues:
1. The search sidebar closes on idle timeout
2. The user interface for Idle Time Limit does not specify the unit in which the timeout is being configured

The combination of these made for a very vexing issue that took a long time to isolate, where the search sidebar was totally unusable due to constantly closing every few milliseconds, making it impossible to search for and contact someone.

### Server Setup Information:

* Version of Rocket.Chat Server: 0.63.1
* Operating System: Ubuntu 16.04 LTS  (4.4.0-119-generic)
* Deployment Method(snap/docker/tar/etc): SNAP
* Number of Running Instances: 1
* DB Replicaset Oplog: Enabled?
* Node Version: v8.9.4
* mongoDB Version: v3.2.7

### Client Binary

* Rocket.Chat+ Version 2.10.5
* Windows 7 Professional x64 Service Pack 1

### Steps to Reproduce:

1. Adjust the automatic-away idle time limit to something really low:
My Account → Preferences → USER PRESENCE → Idle Time Limit: 300
2. Go back to normal messaging, then press the magnifying glass "Search" button to bring up the Search sidebar.
3. Observe that in 300 milliseconds or less, the search bar closes, making it unusable for searching.

### Expected behavior:

The search sidebar does not close.
The Idle Time Limit specifies the unit of time.

### Actual behavior:

The search sidebar closes too quickly for usability.
The Idle Time Limit does not specify the unit of time.

### PR DESCRIPTION
**Pull request by ggazzo**
@RocketChat/core 

Closes #10004
Closes #10415
Closes #9253
Closes #9295 
Closes #5341

### CODE REVIEW

File: packages/rocketchat-ui-sidenav/client/chatRoomItem.js [hunk 1]
```diff
@@ -34,18 +34,27 @@ Template.chatRoomItem.helpers({
...
+
+RocketChat.callbacks.add('enter-room', (sub) => {
+	const items = $('.rooms-list .sidebar-item');
+	items.filter('.sidebar-item--active').removeClass('sidebar-item--active');
+	items.filter(`[data-id=${ sub.rid }]`).addClass('sidebar-item--active');
+	return sub;
+});
+
+Template.sidebarItemStatus.helpers({
+	statusClass() {
```
Comment (sampaiodiego) [inline packages/rocketchat-ui-sidenav/client/chatRoomItem.js]: this is not working as expected.. it is showing `offline` for all users

File: packages/rocketchat-ui-sidenav/client/chatRoomItem.js [hunk 2]
```diff
@@ -34,18 +34,27 @@ Template.chatRoomItem.helpers({
...
+		if (!this.lastMessage && RocketChat.settings.get('Store_Last_Message')) {
+			const room = RocketChat.models.Rooms.findOne(this.rid || this._id, { fields: { lastMessage: 1 } });
+			roomData.lastMessage = room && room.lastMessage || { msg: t('No_messages_yet') };
 		}
-
 		return roomData;
 	}
 });
+
+RocketChat.callbacks.add('enter-room', (sub) => {
```
Comment (sampaiodiego) [inline packages/rocketchat-ui-sidenav/client/chatRoomItem.js]: my channels did not have active indicators while testing

Comment (ggazzo) [reply packages/rocketchat-ui-sidenav/client/chatRoomItem.js]: fixed :)

File: packages/rocketchat-ui-sidenav/client/chatRoomItem.js (hunk 1 above)
Comment (ggazzo) [reply packages/rocketchat-ui-sidenav/client/chatRoomItem.js]: are you sure? 
image

Comment (sampaiodiego) [reply packages/rocketchat-ui-sidenav/client/chatRoomItem.js]: did you test with any other user or just yours?

File: packages/rocketchat-authorization/server/models/Base.js [hunk 3]
```diff
@@ -17,7 +17,7 @@ RocketChat.models._Base.prototype.isUserInRole = function(userId, roleName, scop
 	}
 
 	query.roles = roleName;
-	return !_.isUndefined(this.findOne(query));
+	return !_.isUndefined(this.findOne(query, {fields: {roles: 1}}));
```
Comment (ggazzo) [inline packages/rocketchat-authorization/server/models/Base.js]: amazing

**Comment by sampaiodiego** [comment]
what about adding end-to-end tests to sidenav view mode and sort options?

**Comment by ggazzo** [comment]
image

**Comment by ggazzo** [comment]
\o/

**Pull request merged**